src/main/java/org/maxgamer/quickshop/Listeners/PlayerListener.java

```diff
@@ -251,31 +251,24 @@ public void onInventoryClose(InventoryCloseEvent e) {
             return;
         }
 
-        try {
-            Inventory inventory = e.getInventory();
+        final Inventory inventory = e.getInventory();
+        final Location location = inventory.getLocation();
 
-            // FIXME: 24/11/2019 inventory cannot be null
-            //noinspection ConstantConditions
-            if (inventory == null) {
-                return;
-            }
-
-            final Location location = inventory.getLocation();
-
-            if (location == null) {
-                return;
-            }
-
-            final Shop shop = plugin.getShopManager().getShop(location);
+        if (location == null) {
+            return;
+        }
 
-            if (shop == null) {
-                return;
-            }
+        final Shop shop = plugin.getShopManager().getShop(location);
```

Comment:
final Shop shop = plugin.getShopManager().getShop(location);
Useless code, getShopIncludeAttached contains this, double check will harm a lot performance, because this method performance not very good.
@portlek

Reply:
fixed